miden-lib/src/transaction/inputs.rs

```diff
@@ -93,6 +113,7 @@ fn extend_advice_inputs(
     add_account_to_advice_inputs(tx_inputs.account(), tx_inputs.account_seed(), advice_inputs);
     add_input_notes_to_advice_inputs(tx_inputs.input_notes(), advice_inputs);
     add_tx_script_inputs_to_advice_map(tx_script, advice_inputs);
+    add_note_args_to_advice_map(note_args, advice_inputs);
```

Comment:
Instead of adding this new function, I think we can process note args in `add_input_notes_to_advice_inputs`.